packages/elements/src/slider/index.ts

```diff
@@ -575,29 +581,99 @@ export class Slider extends ControlElement {
   }
 
   /**
-   * Calculate percentage from value, min and max
-   * @param value value to be calculated
-   * @returns percentage
+   * Handles key down event on thumbs
+   * @param event Keyboard event
+   * @returns {void}
    */
-  private calculatePercentage (value: number): number {
-    const percentage = Math.abs((((value || 0) - this.minNumber) / (this.maxNumber - this.minNumber)) * 100);
-    if (percentage > 100) {
-      return 100;
+  private onKeyDown (event: KeyboardEvent): void {
+    if (this.readonly || this.minNumber === this.maxNumber) {
+      return;
+    }
+
+    switch (event.key) {
+      case 'ArrowDown':
+      case 'Down':
+      case 'ArrowLeft':
+      case 'Left':
+        this.onApplyStep(Direction.Down);
+        break;
+      case 'ArrowUp':
+      case 'Up':
+      case 'ArrowRight':
+      case 'Right':
+        this.onApplyStep(Direction.Up);
+        break;
+      default:
+        return;
+    }
+
+    event.preventDefault();
+  }
+
+  /**
+   * Increase or decrease value depending on direction
+   * And fires value change event
+   * @param direction Up or Down
+   * @returns {void}
+   */
+  private onApplyStep (direction: Direction): void {
+    const currentPercentageValue = this.calculatePercentage(this.valueNumber, 1);
+    const percentageStep = this.calculatePercentage(this.minNumber + this.stepRange, 1);
+
+    const percentageValue = direction === Direction.Up ? currentPercentageValue + percentageStep : currentPercentageValue - percentageStep;
+    const nearestPossibleValue = this.getNearestPossibleValue(percentageValue);
+
+    const value = this.getValueFromPercentage(nearestPossibleValue);
+
+    this.value = this.format(value);
+    this.dispatchDataChangedEvent();
+  }
+
+  /**
+   * Get percentage value from decimal fraction number
+   * @param value decimal fraction value
+   * @param multiplier defaults to 100
+   * @returns percentage as a fraction of 100
+   */
+  private calculatePercentage (value: number, multiplier = 100): number {
+    const percentage = Math.abs((((value || 0) - this.minNumber) / (this.maxNumber - this.minNumber)) * multiplier);
+
+    if (percentage > multiplier) {
+      return multiplier;
     }
     else if (percentage < 0) {
       return 0;
     }
-    else {
-      return percentage;
-    }
+
+    return percentage;
   }
 
   /**
-   * Input number event blur and notify property
-   * @param event event blur input number field
+   * Adds active attribute used in styling
+   * @param event focus event
    * @returns {void}
    */
-  private onBlur (event: FocusEvent): void {
+  private onThumbFocus (event: FocusEvent): void {
+    const thumb = event.target as HTMLDivElement;
+    thumb.setAttribute('active', '');
```

Comment:
Here and in other places use the `@state` instead. It improves code visibility and ensures that all internal element updates are happening at one place during `render`.

Reply:
There must be a few `@state` to support multiple thumbs. I find this way easier to read as it is hooked on each thumb automatically.